client/src/pages/profile/Profile.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useMutation, useQueryClient } from 'react-query';
+
+import {
+  Alert,
+  Box,
+  Card,
+  CardContent,
+  Container,
+  Typography,
+} from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+
+import { FormInputField } from '../../components/form-inputs/FormInputField';
+import { StaffUpdateData } from '../../types';
+import Spacer from '../../components/spacer/Spacer';
+import { updateStaff } from '../../data';
+import { useStore } from '../../store';
+import { sanitizeData } from './utils';
+
+const ProfilePage = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [error, setError] = useState('');
+  const { control, handleSubmit, watch } = useForm({});
+  const { user } = useStore();
+  const password = useRef('');
+  const queryClient = useQueryClient();
+
+  password.current = watch('password', '') || '';
```

Comment:
I'm not sure storing the password in a mutable ref is necessary, I would just store the result from `watch` in a const and evaluate it in your JSX. In fact using `watch` directly in your JSX is a pattern supported by our library.

This adds a lot of complexity for performance improvements we may never see.

If you have a particular reason for using a ref, feel free to let me know